src/monodroid/jni/internal-pinvoke-api.cc

```diff
@@ -0,0 +1,333 @@
+}
+
+MONO_API void
+_monodroid_gref_log (const char *message)
+{
+	internal_calls->monodroid_gref_log (message);
+}
+
+MONO_API int
+_monodroid_gref_log_new (jobject curHandle, char curType, jobject newHandle, char newType, const char *threadName, int threadId, const char *from, int from_writable)
+{
+	return internal_calls->monodroid_gref_log_new (curHandle, curType, newHandle, newType, threadName, threadId, from, from_writable);
+}
+
+MONO_API void
+_monodroid_gref_log_delete (jobject handle, char type, const char *threadName, int threadId, const char *from, int from_writable)
+{
+	internal_calls->monodroid_gref_log_delete (handle, type, threadName, threadId, from, from_writable);
+}
+
+MONO_API void
+_monodroid_weak_gref_new (jobject curHandle, char curType, jobject newHandle, char newType, const char *threadName, int threadId, const char *from, int from_writable)
+{
+	internal_calls->monodroid_weak_gref_new (curHandle, curType, newHandle, newType, threadName, threadId, from, from_writable);
+}
+
+MONO_API void
+_monodroid_weak_gref_delete (jobject handle, char type, const char *threadName, int threadId, const char *from, int from_writable)
+{
+	internal_calls->monodroid_weak_gref_delete (handle, type, threadName, threadId, from, from_writable);
+}
+
+MONO_API void
+_monodroid_lref_log_new (int lrefc, jobject handle, char type, const char *threadName, int threadId, const char *from, int from_writable)
+{
+	internal_calls->monodroid_lref_log_new (lrefc, handle, type, threadName, threadId, from, from_writable);
+}
+
+MONO_API void
+_monodroid_lref_log_delete (int lrefc, jobject handle, char type, const char *threadName, int threadId, const char *from, int from_writable)
+{
+	internal_calls->monodroid_lref_log_delete (lrefc, handle, type, threadName, threadId, from, from_writable);
+}
+
+MONO_API void
+_monodroid_gc_wait_for_bridge_processing (void)
+{
+	internal_calls->monodroid_gc_wait_for_bridge_processing ();
+}
+
+/* !DO NOT REMOVE! Used by Mono BCL */
+MONO_API int
+_monodroid_get_android_api_level (void)
+{
+	return internal_calls->monodroid_get_android_api_level ();
+}
+
+/* Can be called by a native debugger to break the wait on startup */
+MONO_API void
+monodroid_clear_gdb_wait (void)
+{
+	internal_calls->monodroid_clear_gdb_wait ();
+}
+
+MONO_API void*
+_monodroid_get_identity_hash_code (JNIEnv *env, void *v)
+{
+	return internal_calls->monodroid_get_identity_hash_code (env, v);
+}
+
+MONO_API void*
+_monodroid_timezone_get_default_id (void)
+{
+	return internal_calls->monodroid_timezone_get_default_id ();
+}
+
+MONO_API void
+_monodroid_counters_dump (const char *format, ...)
+{
+	va_list args;
+	va_start (args, format);
+	internal_calls->dump_counters (format, args);
+	va_end (args);
+}
+
+/* !DO NOT REMOVE! Used by libgdiplus.so */
+MONO_API int
+_monodroid_get_display_dpi (float *x_dpi, float *y_dpi)
+{
+	return internal_calls->monodroid_get_display_dpi (x_dpi, y_dpi);
+}
+
+MONO_API int
```

Comment:
I'm not sure this is viable, which suggests we need a better "cross-reference mechanism".  For example, `set_assemblies_prefix()` is from Embeddinator-4000: 

  * https://github.com/mono/Embeddinator-4000/blob/d844acae5324fd6ac8eaf6c5c9c254b2b6a8a59e/support/mono_embeddinator.c#L56-L63
  * https://github.com/mono/Embeddinator-4000/blob/d844acae5324fd6ac8eaf6c5c9c254b2b6a8a59e/support/java/Runtime.java#L95
  * https://github.com/mono/Embeddinator-4000/blob/d844acae5324fd6ac8eaf6c5c9c254b2b6a8a59e/support/java/android/AndroidImpl.java#L26

It *looks like* Embeddinator-4000 implicitly requires that `monodroid_embedded_assemblies_set_assemblies_prefix()` be located in `monodroid`, because that's the only library it actually loads.

---

We *really* should use GitHub URLs, if at all possible, including commit hash. :-/

---

That said, a cursory investigation shows that Embeddinator-4000 has bit-rotten to the extent that it likely *can't* work with current xamarin-android -- `monodroid` can't be loaded until *after* `libmonosgen-2.0.so` is loaded, plus other dependencies! -- so perhaps it's not worth worrying about at all.

Which really proves that the only way for "Embeddinator-4000" to work reliably is for the core bits to be in xamarin-android, so that they won't get inadvertently broken.

Reply:
Which I suppose is a long way of saying "let's just remove this symbol as well".

Reply:
Isn't this also used by the Desginer?